### DISCUSSION TITLE
Make Maplibre GL JS first class in react-map-gl

### DISCUSSION THREAD

**Discussion by tordans** [maplibre/maplibre#236] [GL JS Feature proposals]
[React-map-gl](https://github.com/visgl/react-map-gl) is the best library to use Maplibre in React. It does support Maplibre GL JS, but only as afterthought for now. For example internally it still uses Mapbox GL JS types, which can cause type errors.

Ideally, it would be possible to use React-Map-GL with Maplibre GL JS direclty without any dependency on Mapbox GL JS.

**Comment by HarelM** [comment]
There is an ongoing discussion about this in the technical steering committee meetings, unfortunately, this is not for us to decide, but rather the maintainers of react-map-gl.
They are leaning towards maplibre, but it might take more time. We do not want to fork this repo currently.
cc: @Pessimistress

**Reply by Pessimistress** [reply]
react-map-gl existed before maplibre and the two libraries were not that different when first forked. Given that I'm the only maintainer at the moment, I cannot commit to upkeeping two separate code bases. TBH I have been considering making maplibre the first-class citizen and dropping Mapbox to the "also compatible" status.

**Reply by Pessimistress** [reply]
Next step RFC: https://github.com/visgl/react-map-gl/issues/2161

**Comment by birkskyum** [comment]
Announcement of react-maplibre:
https://visgl.github.io/react-maplibre/docs/whats-new
